src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -223,13 +223,11 @@ public DetectBoot(DetectBootFactory detectBootFactory, DetectArgumentState detec
     }
 
     private void publishCollectedPropertyValues(PropertyConfiguration propertyConfiguration, EventSystem eventSystem) throws IllegalAccessException {
-        PropertyInfoCollector propertyInfoCollector = new PropertyInfoCollector(propertyConfiguration);
-        List<PropertyInfo> propertyInfo = propertyInfoCollector.collectPropertyInfo(DetectProperties.allProperties(), PropertyInfoCollector.maskPasswordsAndTokensPredicate());
-        Map<String, String> propertyValues = new HashMap<>();
-        propertyInfo.forEach(
-            it -> propertyValues.put(it.getKey(), it.getValue())
-        );
-        eventSystem.publishEvent(Event.PropertyValuesCollected, new PropertyValues(propertyValues));
+        Map<String, String> rawPropertyKeyValues = propertyConfiguration.getRawKeyValueMap(new HashSet<>(DetectProperties.allProperties()));
+        PropertyMasker propertyMasker = new PropertyMasker();
+        Predicate<String> shouldMaskRawValue = propertyKey -> propertyKey.toLowerCase().contains("password") || propertyKey.toLowerCase().contains("api.token") || propertyKey.toLowerCase().contains("access.token");
```

Comment:
The predicate should be shared but I think it appears twice in the code. I don't really mind where it lives, something like DetectPropertyUtil.maskPredicate would be appropriate.

Reply:
I agree it should be shared but I wasn't sure where it would live since configuration project would need access but the predicate itself is somewhat Detect specific.  are you suggesting a DetectPropertyUtil in /configuration?

Reply:
It could go in src/main/configuration.

Reply:
Sure, I guess it makes sense to just pass the predicate into PropertyConfigurationHelpContext.